### PR TITLE
rework of filters to add multivalue functionality

### PR DESCRIPTION
**Pull request by Blackbaud-LuisBello**
and other changes, such as giving the checklist list view a method to select specific items

### CODE REVIEW

File: src/modules/list-view-checklist/list-view-checklist.component.ts [hunk 1]
```diff
@@ -164,6 +164,10 @@ export class SkyListViewChecklistComponent extends ListViewComponent implements
     this.dispatcher.next(new ListSelectedSetItemSelectedAction(item.id, ev.checked));
   }
 
+  public setItemsSelection(items: string[], checked: boolean = true) {
```
Comment (blackbaud-brandonstirnaman) [inline src/modules/list-view-checklist/list-view-checklist.component.ts]: This seems inconsistent with the setItemSelection which uses a ListItemModel.

File: src/modules/list/state/filters/filter-data.model.ts [hunk 2]
```diff
@@ -4,17 +4,33 @@ import * as moment from 'moment';
 export class ListFilterDataModel {
   public id: string = moment().toDate().getTime().toString();
   public value: any = ''; // we use an empty string instead of null to simplify ngModel bindings
+  public displayValue: string = '';
   public onChange: EventEmitter<any> = new EventEmitter<any>();
 
-  public changed(value: any) {
-    this.value = value;
+  public changed(value: any, key?: string, displayValue?: string) {
+    if (key) {
```
Comment (blackbaud-brandonstirnaman) [inline src/modules/list/state/filters/filter-data.model.ts]: Add some comments on what it means when a key is provided

**Comment by Blackbaud-LuisBello** [comment]
updated

File: src/modules/list/state/filters/filter-data.model.ts [hunk 3]
```diff
@@ -4,17 +4,35 @@ import * as moment from 'moment';
 export class ListFilterDataModel {
   public id: string = moment().toDate().getTime().toString();
   public value: any = ''; // we use an empty string instead of null to simplify ngModel bindings
+  public displayValue: string = '';
   public onChange: EventEmitter<any> = new EventEmitter<any>();
 
-  public changed(value: any) {
-    this.value = value;
+  public changed(value: any, key?: string, displayValue?: string) {
+    // key is specified when the filter operator is different than the default operator of 'equals'
```
Comment (blackbaud-joshgerdes) [inline src/modules/list/state/filters/filter-data.model.ts]: That is not what this does exactly.  The 'key' represents the property name on the filter data object that is to be updated. 

We make the assumption that if the key parameter is present then the filter data object (aka this.value) is an object and the value being passed in is the value that should be stored on that object with the property name defined by 'key'.  

It probably makes sense to be a little more checking to assure that this.value is indeed an object before we try and update the property.

**Pull request merged**